### PR TITLE
[Functionalization] Fix cpp tests

### PR DESCRIPTION
**Pull request by wonjoolee95**
Summary:
Lowers ops `copy`, `copy_`, `_prelu_kernel`, and `diag_embed` to fix cpp tests that were failing with functionalization. 

Note that we lower both `copy` and `copy_` ops. When we only lower `copy` op, the in-place version `copy_` op gets auto-generated. However, upstream PyTorch had an assertion that this auto-generated was wrong. To by pass, we're manually lowering `copy_` as well.

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:2323]: Why don't we just replace prelu with _prelu_kernel?

File: torch_xla/csrc/aten_xla_type.cpp
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:1110]: It looks like the TS backend implemented it via passing it to functionalize again, wdyt?: https://github.com/pytorch/pytorch/blob/c29cfe4d02d9d3479961fc1423f87d1fb29aa959/torch/csrc/lazy/ts_backend/ts_native_functions.cpp#L525

File: torch_xla/csrc/elementwise.cpp
Comment (alanwaketan) [inline torch_xla/csrc/elementwise.cpp:231]: Can you explain a little bit on this change?

File: torch_xla/csrc/aten_xla_type.cpp
Comment (wonjoolee95) [reply torch_xla/csrc/aten_xla_type.cpp]: We can definitely do that. Good point, updated.

File: torch_xla/csrc/aten_xla_type.cpp
Comment (wonjoolee95) [reply torch_xla/csrc/aten_xla_type.cpp]: I gave it a try, seems like the tests are still passing. Good catch, updated.

File: torch_xla/csrc/elementwise.cpp
Comment (wonjoolee95) [reply torch_xla/csrc/elementwise.cpp:231]: Now, the `at::Tensor& weight` that is passed to `XLANativeFunctions::_prelu_kernel` already broadcasted. So updated our lowering to remove the unnecessary broadcasting logic. 

Also updated in the PR summary above to include this reasoning.

File: torch_xla/csrc/aten_xla_type.cpp
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:1058]: I was doing the same thing on the other pull request: https://github.com/pytorch/xla/pull/4477. And then it fails some of the CI. I guess this won't be the right approach. Can you maybe exclude this one and then land the remaining changes?

File: torch_xla/csrc/elementwise.cpp
Comment (alanwaketan) [reply torch_xla/csrc/elementwise.cpp:231]: Sounds good to me.

**Review by alanwaketan — APPROVED** [review]
Please double check the copy lowering. Otherwise, LGTM.

File: torch_xla/csrc/aten_xla_type.cpp
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:1061]: Wait a minute. Yea, I think this additional copy_ lowering should be able to fix the issue I'm facing on the other pull request.

I hit an internal assertion on the other pull request that basically suggest the auto-generated copy_ lowering is wrong.

If the CIs are all green, I will let you handle the copy/copy_ lowering in your PR here. And then I will do a rebase after your merge.

File: torch_xla/csrc/aten_xla_type.cpp
Comment (alanwaketan) [inline torch_xla/csrc/aten_xla_type.cpp:1065]: Would it be better to?
```
_copy_from(self, src, non_blocking);
return self;
```

Comment (wonjoolee95) [reply torch_xla/csrc/aten_xla_type.cpp]: Okay, I noticed that `test_copy_noncontig_xla` test was failing with:
```
AssertionError: Tensor-likes are not close!
```
Looking back at this, it should've been `_copy_from(src, self, non_blocking)`, not `_copy_from(self, src, non_blocking)`. After changing this, the `test_copy_noncontig_xla` test is now passing locally. I'll wait for the CI to verify. Now this op looks like:
```
at::Tensor& XLANativeFunctions::copy_(at::Tensor& self, const at::Tensor& src,
                                      bool non_blocking) {
  TORCH_LAZY_FN_COUNTER("xla::");
  _copy_from(src, self, non_blocking);
  return self;
}
```

Comment (alanwaketan) [reply torch_xla/csrc/aten_xla_type.cpp]: Thanks for the update. Can you leave a comment on the commit log to suggest why both copy and copy_ are lowered? It could be simple as we don't want to use the auto-generated copy_ lowering.

Comment (wonjoolee95) [reply torch_xla/csrc/aten_xla_type.cpp]: Sounds good, updated.

**Comment by wonjoolee95** [comment]
CI is all green, will merge this.

**Pull request merged**
